lib/testrail-rspec/update-testrails.rb

```diff
@@ -4,14 +4,20 @@ module TestrailRSpec
   class UpdateTestRails
     attr_accessor :client
 
-    def initialize(scenario)
+    def initialize(scenario, config: {})
       @scenario = scenario
 
       if File.exist? './testrail_config.yml'
         @config = YAML.load_file("./testrail_config.yml")['testrail']
         raise 'TestRail configuration file not loaded successfully' if @config.nil?
-      else
-        raise 'TestRail configuration file is required'
+      end
+
+      if !config.nil? && !config.empty?
+        @config = (@config || {}).merge(config)
```

Comment:
I actually found a bug with this. `config` as proposed in the README has symbol keys, but everywhere `@config` is referenced in this class it is access with string keys. We need to stringify the keys of the passed config object.

This line should be `@config = (@config || {}).merge(config.transform_keys(&:to_s))` or similar.

Reply:
Good catch. Thanks